bin/varnishd/cache/cache_fetch.c

```diff
@@ -580,6 +586,12 @@ vbf_stp_fetchbody(struct worker *wrk, struct busyobj *bo)
 		}
 		AZ(vfc->failed);
 		l = est;
+		/* Limit fetches by pass readahead size */
+		oc = bo->fetch_objcore;
+		if (oc->boc->pass_prefetch > 0 &&
+			l > 2 * oc->boc->pass_prefetch) {
```

Comment:
why `2 *` ?

Reply:
Bugwash:

> 15:04 \< dridi> the only reason i could see would be for a '/2' instead
15:04 \< dridi> up to two segments of half pass_prefetch
15:04 \< slink> I don't see a case for / 2 either
15:04 \< dridi> meaning that client can be delivered a segment while fetch is happening on the next segment
15:05 \< slink> I see the case for / 2 too
15:05 \< dridi> without more than pass_prefetch in flight

Reply:
/me forgot a ;) at the second statement. Dridi convinced me

Reply:
I need to see some kind of agreement between everyone here, especially martin who worked this math out with me last time.

Reply:
@mbgrydeland Dridi and me think this should be `/ 2`?

Reply:
I think this should just be `l`.

In its current form the fetch thread will wait for the client to consume the full segment before continuing.

If we at a later point improve on this with a double-buffered approach that allows fetch and delivery to happen concurrently, then `l/2` would make more sense.

Reply:
Alright, I think that clears up this point